### PR TITLE
deneme

### PR DESCRIPTION
**Pull request by MustafaEmirKocyigitt**

### CODE REVIEW

File: database/migrations/2024_04_21_122749_create_reviews_table.php
Comment (Olgu-Batin-Topal) [inline database/migrations/2024_04_21_122749_create_reviews_table.php:1]:
```php
public function up(): void
    {
Schema::create('books', function (Blueprint $table) {
            $table->id();

            $table->string('title');
            $table->string('author');
            $table->timestamps();
        });

        Schema::create('reviews', function (Blueprint $table) {
            $table->id();

            $table->unsignedBigInteger("book_id");
            $table->text("review");
            $table->unsignedTinyInteger("rating");
            $table->foreign('book_id')->constraint()->cascadeOnDelete();
            $table->timestamps();

           
        });
    }

    /**
     * Reverse the migrations.
     */
    public function down(): void
    {
        Schema::dropIfExists('books');
        Schema::dropIfExists('reviews');
    }
```

Comment (Olgu-Batin-Topal) [reply database/migrations/2024_04_21_122749_create_reviews_table.php:1]: 1. Bunu bu dosyaya yapıştır.
2. Migrations dosyalarından books'a ait olanı sil.

**Review by Olgu-Batin-Topal — CHANGES_REQUESTED** [review]

File: database/migrations/2024_04_21_122749_create_reviews_table.php [hunk 1]
```diff
@@ -11,13 +11,20 @@
...
+            $table->timestamps();
+        });
         Schema::create('reviews', function (Blueprint $table) {
             $table->id();
 
             $table->unsignedBigInteger("book_id");
             $table->text("review");
             $table->unsignedTinyInteger("rating");
-            $table->foreign('book_id')->references('id')->on('books')->cascadeOnDelete();
+            $table->foreign('book_id')->constraint()->cascadeOnDelete();
```
Comment (Olgu-Batin-Topal) [inline database/migrations/2024_04_21_122749_create_reviews_table.php]:
```php
$table->foreign('book_id')->references('id')->on('books')->cascadeOnDelete();
```

Comment (Olgu-Batin-Topal) [reply database/migrations/2024_04_21_122749_create_reviews_table.php]: Bu şekilde revize eder misin?

**Review by Olgu-Batin-Topal — CHANGES_REQUESTED** [review]

File: database/migrations/2024_04_21_122749_create_reviews_table.php (hunk 1 above)
Comment (MustafaEmirKocyigitt) [reply database/migrations/2024_04_21_122749_create_reviews_table.php]: Dropping all tables ................................................................................................................. 39.57ms DONE

   INFO  Preparing database.  

  Creating migration table ............................................................................................................. 7.01ms DONE

   INFO  Running migrations.  

  0001_01_01_000000_create_users_table ................................................................................................ 31.15ms DONE
  0001_01_01_000001_create_cache_table ................................................................................................. 8.14ms DONE
  0001_01_01_000002_create_jobs_table ................................................................................................. 20.96ms DONE
  2024_04_21_122749_create_reviews_table .............................................................................................. 12.65ms FAIL

   Illuminate\Database\QueryException 

  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') on delete cascade' at line 1 (Connection: mysql, SQL: alter table `reviews` add constraint `reviews_book_id_foreign` foreign key (`book_id`) references `` () on delete cascade)

  at vendor\laravel\framework\src\Illuminate\Database\Connection.php:813
    809▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    810▕                 );
    811▕             }
    812▕
  ➜ 813▕             throw new QueryException(
    814▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    815▕             );
    816▕         }
    817▕     }

  1   vendor\laravel\framework\src\Illuminate\Database\Connection.php:565
      PDOException::("SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') on delete cascade' at line 1")

  2   vendor\laravel\framework\src\Illuminate\Database\Connection.php:565
      PDO::prepare("alter table `reviews` add constraint `reviews_book_id_foreign` foreign key (`book_id`) references `` () on delete cascade")


HATA DONDU

**Pull request merged**